### PR TITLE
Can now pass a facts hash for classification

### LINKED ISSUE #10: It does not appear possible to post facts to the classification object
As a user, I would like to be able to post facts to the classification API to test classification rules.  The current implementation provides no mechanism to pass the facts and trusted hashes into the classification method.

### PR DESCRIPTION
**Pull request by binford2k**
Ping @mhjacks

Fixes #10

### CODE REVIEW

**Comment by mhjacks** [comment]
Thanks!  I'll give this a workout on Monday.

File: README.md [hunk 1]
```diff
@@ -59,6 +59,55 @@ group_delta = {"variables"=>{"key"=>"value"}, "id"=>my_group_id, "classes"=>{"mo
...
+require 'json'
+
+facts = { 'fact' => JSON.parse(`facter -j`) }
+```
+
+Once you have the facts, retrieving classification of a node is simple:
+
+```ruby
+#! /opt/puppetlabs/puppet/bin/ruby
+require 'facter'
```
Comment (briancain) [inline README.md:91]: I'm trying this line within PE, and I seem to get this error:

```
[root@and1y2ph16l587k puppet-classify]# ruby testscript.rb
/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:53:in `require': cannot load such file -- facter (LoadError)
    from /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:53:in `require'
```

I'm guessing this has something to do with where Facter exists within PE?

**Comment by briancain** [comment]
@binford2k thanks so much for the PR! I left a comment, but I'm trying it out as we speak.

**Comment by briancain** [comment]
Just tried this in pry, and it looks to work fine :+1: Let me know what you think about the require bit...and then I'd be happy to merge and ship a new version of this gem. Thanks!

**Comment by mhjacks** [comment]
Thanks a ton, both, for looking at this and working on it.  I'm wondering, should there be some more guidance about how to pass facts in?  I'm working with a large number of files that were generated by puppetserver 1.x for example, and they have the YAML tag !ruby/object:Puppet::Node::Facts on them; I'm assuming that "values" would be the thing that gets sent as "fact" in the API call.  There is also a _timestamp attribute in values that probably won't cause trouble but I suspect is filtered somehow.

**Comment by briancain** [comment]
@mhjacks that's a good question. From what I can tell on the [docs](https://docs.puppetlabs.com/pe/latest/nc_classification.html#request-format), you need `keys` and `values` for the set of Facts that you are sending to the classifier api. Does that object also have a `keys` associated with it? If so then I think you should be good sending the entire hash to the API as is.

**Comment by mhjacks** [comment]
No "keys", only "values":

--- !ruby/object:Puppet::Node::Facts
  values:
    mtu_eth1: "1500"
    ...
  expiration: 2016-01-20 13:25:42.529379 +00:00
  name: <a typical FQDN in our environment>

**Comment by briancain** [comment]
@mhjacks huh...it kind of looks like `values` is a key/value hash? Where the key in values is `mtu_eth1` and the value is `"1500"`.

**Comment by mhjacks** [comment]
Yes, that's right.  All the normal facts (and our custom ones, which I'd rather not post here) are there, under the "values" section of the YAML file

**Comment by briancain** [comment]
@mhjacks :+1: great! Yeah I am guessing then you could just use that as your facts hash to send to the classifier service.

File: README.md (hunk 1 above)
Comment (binford2k) [inline README.md:91]: Yeah, you'll have to set the GEMPATH or just use /opt/puppetlabs/puppet/bin/ruby

Comment (briancain) [inline README.md:91]: Cool, yeah that makes sense. Thanks for adding an example!

**Pull request merged**